src/main/java/org/codinjutsu/tools/jenkins/view/util/BuildStatusIcon.java

```diff
@@ -55,7 +65,9 @@ public static JComponent createIcon(boolean combine, BuildStatusAggregator aggre
         if (combine) {
             final JPanel combined = new JPanel();
             combined.setLayout(new BoxLayout(combined, BoxLayout.X_AXIS));
-            icons.forEach(combined::add);
+
+            // Sorted to put the running icon at the front
+            icons.stream().sorted(Comparator.comparing(i -> i.icon != JenkinsControlIcons.Job.GREY)).forEach(combined::add);
             return combined;
         } else {
             return icons.get(0);
```

Comment:
Why this?
You could use to get this result with
```java
        createIcon(icons, aggregator.getRunningBuilds(), buildStatusRenderer, BuildStatusEnum.RUNNING, "running");
        createIcon(icons, aggregator.getBrokenBuilds(), buildStatusRenderer, BuildStatusEnum.FAILURE, "broken");
        createIcon(icons, aggregator.getUnstableBuilds(), buildStatusRenderer, BuildStatusEnum.UNSTABLE, "unstable");
        createIcon(icons, aggregator.getSucceededBuilds(), buildStatusRenderer, BuildStatusEnum.SUCCESS, "succeeded");
```
Wand what Icon should be displayed if 'Show all Build' is disabled? The running or the Failure (or highest status)

Reply:
If Running should shown as single status  this will be ok in my opinion

Reply:
Like I mentioned before, in my opinion it should be:
Single icon priority: Failed, Unstable, Running, Success
Combined order: Running, Failed, Unstable, Success

Running has a higher order than success because it means that something is happening ( and otherwise it would probably never be shown) but it doesn't require an action (like failed or unstable do).

If you disagree I can change it to the way you would like it but for me this makes most sense :).

Reply:
> Single icon priority: Failed, Unstable, Running, Success
Combined order: Running, Failed, Unstable, Success

make sense.,

Reply:
so waht would if we implement an logic for combined:
```java
        if (combine) {
            createIcon(icons, aggregator.getRunningBuilds(), buildStatusRenderer, BuildStatusEnum.RUNNING, "running");
            createIcon(icons, aggregator.getBrokenBuilds(), buildStatusRenderer, BuildStatusEnum.FAILURE, "broken");
            createIcon(icons, aggregator.getUnstableBuilds(), buildStatusRenderer, BuildStatusEnum.UNSTABLE, "unstable");
            createIcon(icons, aggregator.getSucceededBuilds(), buildStatusRenderer, BuildStatusEnum.SUCCESS, "succeeded");
            final JPanel combined = new JPanel();
            combined.setLayout(new BoxLayout(combined, BoxLayout.X_AXIS));
            icons.forEach(combined::add);
            return combined;
        } else {
            // determine highest status and call 
            return createIcon(highestStatusNumber, buildStatusRenderer, highestBuildStatusEnum, highestBuildStatusTooltip);
        }
```

Reply:
Then we  also not need the iconsList and should add them directly to panel. Or what do you think

Reply:
Seeing that it would need another createIcon method with additional logic and probably the same lines that are in the `combine=true` branch with some if statements around them, I think that adds more complexity than the one 'sorted' call.

This one might work as well but I think I still prefer the one-liner over this:
```java
       if (combine) {
            createIcon(icons, aggregator.getRunningBuilds(), buildStatusRenderer, BuildStatusEnum.RUNNING, "running");
            createIcon(icons, aggregator.getBrokenBuilds(), buildStatusRenderer, BuildStatusEnum.FAILURE, "broken");
            createIcon(icons, aggregator.getUnstableBuilds(), buildStatusRenderer, BuildStatusEnum.UNSTABLE, "unstable");
            createIcon(icons, aggregator.getSucceededBuilds(), buildStatusRenderer, BuildStatusEnum.SUCCESS, "succeeded");
            final JPanel combined = new JPanel();
            combined.setLayout(new BoxLayout(combined, BoxLayout.X_AXIS));
            icons.forEach(combined::add);
            return combined;
        } else {
            createIcon(icons, aggregator.getBrokenBuilds(), buildStatusRenderer, BuildStatusEnum.FAILURE, "broken");
            createIcon(icons, aggregator.getUnstableBuilds(), buildStatusRenderer, BuildStatusEnum.UNSTABLE, "unstable");
            createIcon(icons, aggregator.getRunningBuilds(), buildStatusRenderer, BuildStatusEnum.RUNNING, "running");
            createIcon(icons, aggregator.getSucceededBuilds(), buildStatusRenderer, BuildStatusEnum.SUCCESS, "succeeded");
            return icons.get(0);
        }
```

Reply:
yes the one line look cleaner. I have an idea. let me test it shortly

Reply:
Tried to use an enum renderer and then 
```java

        if (combine) {
            order = EnumSet.of(BuildStatusEnum.RUNNING, BuildStatusEnum.FAILURE, BuildStatusEnum.UNSTABLE, BuildStatusEnum.SUCCESS);
        } else {
            order = EnumSet.of(BuildStatusEnum.FAILURE, BuildStatusEnum.UNSTABLE, BuildStatusEnum.RUNNING, BuildStatusEnum.SUCCESS);
        }
        final Stream<BuildStatusIcon> buildStatusIconStream = order.stream().map(status -> ICON_CREATOR_BY_BUILD_STATUS.getOrDefault(status, CREATE_NO_ICON))
                .map(iconCreator -> iconCreator.getIcon(buildStatusRenderer, aggregator))
                .filter(Objects::nonNull);
```

But his seem mor complivated. so your solution ´with sorting is fine for this